cmd/sonobuoy/app/sonobuoyconfig.go

```diff
@@ -41,11 +42,15 @@ func (c *SonobuoyConfig) String() string {
 // Type is needed for pflag.Value.
 func (c *SonobuoyConfig) Type() string { return "Sonobuoy config" }
 
-// Set attempts to read a file, then deserialise the json into a config.Config struct.
+// Set attempts to read a file, then deserialize the json into a config.Config struct.
 func (c *SonobuoyConfig) Set(str string) error {
+	if !reflect.DeepEqual(c.Config, *config.New()) {
```

Comment:
These 4 lines replace a 50 line method and 200+ lines of tests. Extreme simplification.